### PR TITLE
Feature user crud controller

### PR DESCRIPTION
**Pull request by AndreyWolox**
## Summary
- Add user crud controller
- Add error user enum
## Trello Card
https://trello.com/c/dAXR5dRU/21-crear-user-controller-con-las-operaciones-crud-de-user

### CODE REVIEW

File: src/main/java/wolox/training/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        findOne(id);
+        userRepository.deleteById(id);
+    }
+
+    /**
+     * Update user by Id
+     *
+     * @param id:   User identifier (Long)
+     * @param user: User to update(Book)
+     *
```
Comment (CarlosLugo22) [inline src/main/java/wolox/training/controllers/UserController.java:94]: Can add exceptions to documentation like said in the other controller

**Review by CarlosLugo22 — COMMENTED** [review]
Have in mind the comments from Leandro at the other controller

File: src/main/java/wolox/training/controllers/UserController.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+    @Autowired
+    BookRepository bookRepository;
+
+    /**
+     * Find all users
+     *
+     * @return all users
+     */
+    @GetMapping()
```
Comment (leandrocostawolox) [inline src/main/java/wolox/training/controllers/UserController.java]: There is no need to add the extra `()`. You can use just the annotation `@GetMapping`

File: src/main/java/wolox/training/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    /**
+     * Delete user by Id
+     *
+     * @param id: User identifier (Long)
+     */
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable long id) {
+        findOne(id);
+        userRepository.deleteById(id);
```
Comment (leandrocostawolox) [inline src/main/java/wolox/training/controllers/UserController.java:86]: As I commented for the Books controller: `204` will be a better HTTP status code for this endpoint :)

File: src/main/java/wolox/training/controllers/UserController.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+    /**
+     * Add a book to the user
+     *
+     * @param id:     User identifier (Long)
+     * @param bookId: Book identifier (Long)
+     *
+     * @return User whose book was updated
+     */
+    @PutMapping("/{id}/book/{bookId}")
```
Comment (leandrocostawolox) [inline src/main/java/wolox/training/controllers/UserController.java]: `.../books/...` As a convention we should use the resources in their plural form

File: src/main/java/wolox/training/controllers/UserController.java [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+    /**
+     * Remove a book to the user
+     *
+     * @param id:     User identifier (Long)
+     * @param bookId: Book identifier (Long)
+     *
+     * @return User who had the book removed
+     */
+    @DeleteMapping("/{id}/book/{bookId}")
```
Comment (leandrocostawolox) [inline src/main/java/wolox/training/controllers/UserController.java]: Same here: `books` instead of book

**Review by leandrocostawolox — CHANGES_REQUESTED** [review]
Great! Just some minor comments :)

**Review by CarlosLugo22 — APPROVED** [review]

**Review by leandrocostawolox — APPROVED** [review]
Excellent!

**Pull request merged**